jax_unirep/featurize.py

```diff
@@ -3,7 +3,70 @@
 import numpy as np
 
 from .layers import mlstm1900
-from .utils import get_embeddings, load_params_1900
+from .utils import batch_sequences, get_embeddings, load_params_1900
+
+
+def rep_same_lengths(
+    seqs: List[str],
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    This function generates representations of protein sequences that have the same length,
+    by passing them through the UniRep mLSTM.
+
+    :param seqs: A list of same length sequences as strings.
+        If passing only a single sequence, it also needs to be passed inside a list.
+    :returns: A tuple of np.arrays containing the reps.
+        Each `np.array` has shape (n_sequences, 1900).
+    """
+
+    params = load_params_1900()
+    embedded_seqs = get_embeddings(seqs)
+
+    h_final, c_final, h = mlstm1900(params, embedded_seqs)
+    h_avg = h.mean(axis=1)
+
+    return np.array(h_final), np.array(c_final), np.array(h_avg)
+
+
+def rep_arbitrary_lengths(
+    seqs: List[str],
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    This function generates representations of protein sequences of arbitrary length,
+    by batching together all sequences of the same length and passing them through
+    the mLSTM. Original order of sequences is restored in the final output.
```

Comment:
I would have thought of using dictionaries to do this, but on second thought, yours is better because of the preservation of order.

Reply:
I'm not happy with the code triplication though.. probably there are better ways of doing it while preserving order.

Reply:
All good, at least the triplication is wrapped inside a single function and isn't rampant everywhere :smile_cat:.